Packet++/src/SomeIpLayer.cpp

```diff
@@ -286,20 +251,20 @@ void SomeIpLayer::parseNextLayer()
 	uint8_t *payload = m_Data + headerLen;
 	size_t payloadLen = m_DataLen - headerLen;
 
-	/* Check if the header is a valid SomeIpSd header */
-	if (isSomeIpSdPacket(getServiceID(), getMethodID()))
-	{
-		m_NextLayer = new PayloadLayer(payload, payloadLen, this, m_Packet);
-	}
-	/* Check if the payload is a valid SomeIp TP layer */
-	else if (isSomeIpTpPacket(getMessageTypeAsInt()))
+	if (SomeIpLayer::isDataValidSd(payload, payloadLen))

```

Comment:
if we implement the `parseSomeIpLayer(...)`  I spoke about in one of the previous comments, we can probably use it here too